modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/artifacts/JaggeryApplicationGhostDeploymentTestCase.java

```diff
@@ -0,0 +1,496 @@
+
+                public void run() {
+                    HttpResponse httpResponse = null;
+                    try {
+                        httpResponse = HttpURLConnectionClient.sendGetRequest(tenant1JaggApp1Url, null);
+                    } catch (IOException e) {
+                        log.error("Error  when sending a  get request  for :" + tenant1JaggApp1Url, e);
+                    }
+                    synchronized (this) {
+                        String responseDetailedInfo;
+                        String responseData;
+                        if (httpResponse != null) {
+                            responseDetailedInfo = "Response Data :" + httpResponse.getData() +
+                                    "\tResponse Code:" + httpResponse.getResponseCode();
+                            responseData = httpResponse.getData();
+                        } else {
+                            responseDetailedInfo = "Response Data : NULL Object return from HttpURLConnectionClient";
+                            responseData = "NULL Object return";
+                        }
+
+                        responseDataList.add(responseData);
+                        log.info(responseDetailedInfo);
+                        responseDetailedInfoList.add(responseDetailedInfo);
+                    }
+                }
+
+            });
+
+        }
+
+        executorService.shutdown();
+        executorService.awaitTermination(5, TimeUnit.MINUTES);
+        log.info("Concurrent invocation End");
+
+        int correctResponseCount = 0;
+        for (String responseData : responseDataList) {
+            if (JAGG_APP1_RESPONSE.equals(responseData)) {
+                correctResponseCount += 1;
+            }
+
+        }
+        StringBuilder allDetailResponseStringBuffer = new StringBuilder();
+        allDetailResponseStringBuffer.append("\n");
+
+        for (String responseInfo : responseDetailedInfoList) {
+            allDetailResponseStringBuffer.append(responseInfo);
+            allDetailResponseStringBuffer.append("\n");
+        }
+        String allDetailResponse = allDetailResponseStringBuffer.toString();
+
+        webAppStatusTenant1WebApp1 = getWebAppStatus(TENANT_DOMAIN_1, JAGGERY_APP_NAME1);
+        assertEquals(webAppStatusTenant1WebApp1.getTenantStatus().isTenantContextLoaded(), true,
+                " Tenant Context is not loaded. Tenant:" + TENANT_DOMAIN_1);
+        assertEquals(webAppStatusTenant1WebApp1.isWebAppStarted(), true, "Web-App: " + JAGGERY_APP_NAME1 +
+                " is not started in Tenant:" + TENANT_DOMAIN_1);
+        assertEquals(webAppStatusTenant1WebApp1.isWebAppGhost(), false, "Web-App: " + JAGGERY_APP_NAME1 +
+                " is in ghost mode after invoking in Tenant:" + TENANT_DOMAIN_1);
+
+        assertEquals(correctResponseCount, CONCURRENT_THREAD_COUNT, allDetailResponse + "All the concurrent requests" +
+                " not get correct response.");
+
+
+    }
+
+
+    @AfterClass
```

Comment:
It is always recommended to set @AfterClass(alwaysRun = true) for ensuring that this runs irrespective of what groups it belongs to.